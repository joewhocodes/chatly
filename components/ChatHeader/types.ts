export interface ChatHeaderProps {
	navigation: any;
	chatName: string;
	setShowModal: React.Dispatch<React.SetStateAction<boolean>>;
	setShowBlock: React.Dispatch<React.SetStateAction<boolean>>;
}
