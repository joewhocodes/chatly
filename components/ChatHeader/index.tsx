import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Box, Center, Heading, Text, Flex, Image } from 'native-base';

import { ChatHeaderProps } from './types';

const ChatHeader = ({ navigation, chatName, setShowModal }: ChatHeaderProps) => {
	const handleBackPress = () => {
		navigation.navigate('ChatList');
	};

	const handleChangeName = () => {
		setShowModal(true);
	};

	return (
		<Box backgroundColor={'primary.500'}>
			<Box mt={'20px'}>
				<Flex direction='row' justifyContent={'space-between'}>
					<Box>
						<TouchableOpacity onPress={handleBackPress}>
							<Image
								source={require('../../assets/back-arrow.png')}
								alt={'back arrow'}
								mt='7px'
							/>
						</TouchableOpacity>
					</Box>
					<Image
						source={require('../../assets/logo.png')}
						alt={'logo'}
						mr={'45px'}
					/>
					<Text></Text>
				</Flex>
				<Center>
					<Heading color={'white'} fontFamily={'Jua-Regular'}>
						{chatName}
					</Heading>
					<TouchableOpacity onPress={handleChangeName}>
						<Text color='white' fontFamily={'Jua-Regular'}>
							Tap to change name
						</Text>
					</TouchableOpacity>
				</Center>
			</Box>
		</Box>
	);
};

export default ChatHeader;
