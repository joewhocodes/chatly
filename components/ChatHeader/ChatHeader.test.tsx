import React from 'react';
import renderer from 'react-test-renderer';
import { NativeBaseProvider } from 'native-base';

import ChatHeader from './index';

describe('ChatHeader', () => {
  it('renders ChatHeader correctly', () => {
    const chatName = 'Test Chat';
    const setShowModal = jest.fn();

    const tree = renderer.create(
      <NativeBaseProvider>
        <ChatHeader chatName={chatName} setShowModal={setShowModal} navigation={undefined} />
      </NativeBaseProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});