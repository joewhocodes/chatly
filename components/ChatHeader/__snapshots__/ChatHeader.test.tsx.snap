// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader renders ChatHeader correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`ChatHeader renders ChatHeader correctly 2`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
